### PR TITLE
add comments in check_expr_field

### PR DESCRIPTION
**Pull request by oriongonza**
Nothing special, just a few comments and a couple of small cleanups.

### CODE REVIEW

File: compiler/rustc_hir_typeck/src/expr.rs [hunk 1]
```diff
@@ -2748,6 +2761,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 _ => {}
             }
         }
+
+        // We failed to check the expression, report an error.
```
Comment (compiler-errors) [inline compiler/rustc_hir_typeck/src/expr.rs:2765]: This comment doesn't really describe what this is doing. It's attempting to emit an error if we end up derefing to an infer var, like calling `.field` on a base type of `&_`.

**Review by compiler-errors — APPROVED** [review]
Please fix the comment

**Comment by compiler-errors** [comment]
Then r=me

File: compiler/rustc_hir_typeck/src/expr.rs (hunk 1 above)
Comment (dev-ardi) [reply compiler/rustc_hir_typeck/src/expr.rs:2765]: That comment was more general, as in "ok we're no longer in the happy path"

Comment (compiler-errors) [reply compiler/rustc_hir_typeck/src/expr.rs:2765]: Well it's useful regardless to document the random structural resolve call. Pls do that :)

**Comment by dev-ardi** [comment]
r=compiler-errors

**Pull request merged**